<?xml version="1.0" ?>

<launch>
	<!-- JetHexa Compatibility -->
	<arg name="master_name" default="$(env MASTER_NAME)"/>
	<arg name="robot_name" default="$(env ROBOT_NAME)" />
	<arg name="tf_prefix" default="$(arg robot_name)" />
	<arg name="waypoints" default="waypoints" />	
	<arg name="base_frame_id" default="$(arg tf_prefix)/base_link"/>
	<arg name="map_frame_id" default="$(arg tf_prefix)/map"/>
	<arg name="map" default="lv_room" />
	<arg name="map_file" default="$(find uav_follower)/maps/$(arg map).yaml" />
	<arg name="test_mode" default="false" />
	<arg name="record_data" default="false" />
	<arg name="rviz" default="false" />

    <!-- Begin Camera Machinery -->
	<include file="$(find jethexa_peripherals)/launch/camera.launch"></include>
	<!-- JetHexa Navigation Nodes -->
	<group unless="$(arg test_mode)">
		<include file="$(find jethexa_navigation)/launch/jethexa_load_map.launch">
			<arg name="map_file" value="$(arg map_file)" />
		</include> 
		
		<include file="$(find uav_follower)/launch/include/jethexa_navigation.launch">
			<arg name="clicked_point" value="$(arg waypoints)" />
		</include>
	</group>
	
	<include file="$(find jethexa_navigation)/launch/jethexa_navigation_rviz.launch" if="$(arg rviz)"></include>
	<!-- uav_follower Nodes--> 	
	<group ns="uav_follower">
		<rosparam command="load" file="$(find uav_follower)/config/uav_follower_params.yaml" />
        <param name="test_mode" type="bool" value= "$(arg test_mode)" />

		<node name="ss00_Liaison" pkg="uav_follower" type="ss00_Liaison.py" output="screen" required="true">
			<param name="move_base_result" value="/$(arg robot_name)/move_base/result"/>
			<param name="waypoints" value="/$(arg robot_name)/$(arg waypoints)" if="$(arg test_mode)"/>	<!-- Note that it's a global name -->
		</node>
        
		<node name="ss01_Photographer" pkg="uav_follower" type="ss01_Photographer.py" output="screen" required="true"></node>
        
		<node name="ss02_Detector" pkg="uav_follower" type="ss02_Detector.py" output="screen" required="true"></node>
        
		<node name="ss03_DataProcessor" pkg="uav_follower" type="ss03_DataProcessor.py" output="screen" required="true">
			<param name="frame_id" value="$(arg map_frame_id)" />
			<param name="waypoints" value="/$(arg robot_name)/$(arg waypoints)" />	<!-- Note that it's a global name -->
		</node>

		<node name="ss04_tf2Watcher" pkg="uav_follower" type="ss04_tf2Watcher.py" output="screen" required="true" unless="$(arg test_mode)">
			<param name="base_frame" value="$(arg base_frame_id)" />
			<param name="map_frame" value="$(arg map_frame_id)" />
		</node>

		<node name="array_saver" pkg="uav_follower" type="array_saver.py" output="screen" respawn="true" if="$(arg test_mode)">
			<param name="log_dir" type="string" value="$(find uav_follower)/logs" />
		</node>

		<node name="data_recorder" pkg="uav_follower" type="rosbag_script.zsh" if="$(arg record_data)"></node>
	</group>
</launch>


